src/libraries/System.Text.Json/gen/Reflection/TypeWrapper.cs

```diff
@@ -207,20 +223,53 @@ public override bool IsEnum
 
         public override bool IsGenericType => _namedTypeSymbol?.IsGenericType == true;
 
-        public override bool ContainsGenericParameters => _namedTypeSymbol?.IsUnboundGenericType == true;
+        public override bool ContainsGenericParameters
+        {
+            get
+            {
+                if (IsGenericParameter)
+                {
+                    return true;
+                }
+                for (INamedTypeSymbol currentSymbol = _namedTypeSymbol; currentSymbol != null; currentSymbol = currentSymbol.ContainingType)
+                {
+                    if (currentSymbol.TypeArguments.Any(arg => arg.TypeKind == TypeKind.TypeParameter))
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
```

Comment:
this seems pretty heavy for property. Consider caching this value after the first usage or on initialization if it's used more than once

Reply:
I agree, it seems heavy =)
I didn't introduce a cache here because:
- I don't want to affect memory usage (it seems that many TypeWrappers are allocated during generation);
- I assume that in most cases currentSymbol.ContainingType == null, TypeArguments is empty or contains few ellements.